src/Autofac.Extras.AggregateService/ResolvingInterceptor.cs

```diff
@@ -117,15 +119,45 @@ private void MethodWithoutParams(IInvocation invocation)
                 if (parameters.Length > 0)
                 {
                     // Methods with parameters
-                    methodMap.Add(method, invocation =>
-                        {
-                            var arguments = invocation.Arguments;
-                            var typedParameters = parameters
-                                .Select(info => (Parameter)new TypedParameter(info.ParameterType, arguments[info.Position]));
-
-                            // To handle open generics, this resolves the return type of the invocation rather than the scanned method.
-                            invocation.ReturnValue = _context.Resolve(invocation.Method.ReturnType, typedParameters);
-                        });
+                    if (parameters.Any(p => p.ParameterType.IsGenericType))
+                    {
+                        // There are some open generic parameters so resolve a
+                        // Func<X, Y> corresponding to the method parameters and
+                        // method return type. Core Autofac will handle the type
+                        // mapping from open generic to closed generic, etc.
+                        methodMap.Add(
+                            method,
+                            invocation =>
+                            {
+                                var targetMethod = invocation.Method;
+                                var funcArgTypes = targetMethod.GetParameters().OrderBy(p => p.Position).Select(p => p.ParameterType).Append(targetMethod.ReturnType).ToArray();
```

Comment:
Minor perf suggestion (given the reflection going on here), but `ToList` is generally more performant than `ToArray` for a variable number of elements.